File: two_pointers/find_all_anagrams_in_a_string.py
class Solution:
    def findAnagrams(self, s: str, p: str) -> List[int]:
        left = 0
        right = len(p) - 1
        match_indexes = []

        while right <= len(s) - 1:
            if sorted(s[left:right + 1]) == sorted(p):
                match_indexes.append(left)
            left += 1
            right += 1
            
        return match_indexes

File: two_pointers/find_all_anagrams_in_a_string.rb
# @param {String} s
# @param {String} p
# @return {Integer[]}
def find_anagrams(s, p)
    match_indexes = []
    
    left = 0
    right = p.size - 1
    while right <= s.size
        if s[left..right].chars.sort == p.chars.sort
            match_indexes << left
        end
        left += 1
        right += 1
    end

    match_indexes
end